src/solution.c

```diff
@@ -762,7 +762,7 @@ void process_matched_obs(u8 n_sds, gps_time_t *t, sdiff_t *sds, u16 base_id)
   }
 }
 
-static THD_WORKING_AREA(wa_time_matched_obs_thread, 20000);
+static THD_WORKING_AREA(wa_time_matched_obs_thread, 18000);
```

Comment:
Remove this change. The heap issue is likely not related to this PR, but we need to fix it separately and in a less hacky manner.